### ISSUE TITLE
Database scaffolding produces empty method call and no ValueGeneratedOn*() call when using datetime/timestamp columns

### ISSUE THREAD

**Issue by Slushnas** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#877]
### Steps to reproduce
Create the table:
```
create table TimestampTable(
    Id int not null,
    Ts timestamp default '0000-00-00 00:00:00' on update CURRENT_TIMESTAMP,
    Dt datetime default '0000-00-00 00:00:00' on update CURRENT_TIMESTAMP,
    primary key (Id)
);
```
Run the scaffolder:
```
dotnet ef dbcontext scaffold 'ConnectionString' Pomelo.EntityFrameworkCore.MySql -o Test
```

### The issue
The scaffolder produces the following:
```
protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<TimestampTable>(entity =>
            {
                entity.Property(e => e.Id).HasColumnType("int(11)");

                entity.Property(e => e.Dt)
                    .HasColumnType("datetime")
                    .HasDefaultValueSql("'0000-00-00 00:00:00'")
                    .();

                entity.Property(e => e.Ts)
                    .HasColumnType("timestamp")
                    .HasDefaultValueSql("'0000-00-00 00:00:00'")
                    .();
            });

            OnModelCreatingPartial(modelBuilder);
        }

        partial void OnModelCreatingPartial(ModelBuilder modelBuilder);
}
```
**Issue 1:** The default value strings have added single quotes which may be a problem but that behavior is already detailed in Issue #703.

**Issue 2:** The datetime and timestamp properties have an empty method call `.()` appended which causes build errors.

**Issue 3:** I would expect the scaffolder to append .ValueGeneratedOnUpdate() or .ValueGeneratedOnAddOrUpdate() to the property definition based on the syntax here: https://dev.mysql.com/doc/refman/8.0/en/timestamp-initialization.html. Perhaps there is a bug with this functionality that is causing issue 2?

Also - I'm not sure if the OnModelCreatingPartial() method is supposed to exist here or not as it doesn't appear to be used.

### Further technical details

MySQL version: **10.1.41-MariaDB-0ubuntu0.18.04.1**
Operating system: **Ubuntu 18.04**
Pomelo.EntityFrameworkCore.MySql version: **3.0.0-rc1.final**
Microsoft.AspNetCore.App version: **3.0**

**Comment by lauxjpn** [comment]
Issue 2 & 3 are likely to be related.

> Also - I'm not sure if the OnModelCreatingPartial() method is supposed to exist here or not as it doesn't appear to be used.

The point of this method is, that you can implement it in a partial class. If you implement it, it will be called as part of `OnModelCreating()`. Otherwise, it will not be called.

**Comment by Slushnas** [comment]
Thanks, I had wondered if it was there as a placeholder in case it was needed.

**Comment by lauxjpn** [comment]
Issue 1 is not an issue, but the correct behavior. Issue #703 is just a special case, where single quotes are invalid.

For issue 2/3, I opened https://github.com/aspnet/EntityFrameworkCore/issues/18579 upstream, as this seems to be a bug in EF Core.

**Comment by mguinness** [comment]
The upstream bug is fixed and will be in the 3.1 release.
